### PR TITLE
Fix google search

### PR DESCRIPTION
**Pull request by TPGamesNL**
See #7

In progress. Description will come later.

### CODE REVIEW

File: api.py [hunk 1]
```diff
@@ -79,4 +79,5 @@ def signal_handler(sig, frame):
 signal.signal(signal.SIGINT, signal_handler)
 
 # Start Flask app, bind to all interfaces
-app.run(host="0.0.0.0")
+# app.run(host="0.0.0.0")
+print(detection.test('https://www.paypal.com/signin', 'https://www.paypal.com/signin', 'd19a51e9-f4dc-425e-b666-e2429403613e', 'Log in to your PayPal account', None))
```
Comment (TPGamesNL) [inline api.py:83]: Note that this will (obviously) be removed before merge

File: engines/google.py [hunk 2]
```diff
@@ -138,6 +128,24 @@ def find_matches(self) -> list:
...
+        matches = self.search_html.find('.egMi0.kCrYT')
+        for match in matches:
+            for l in match.absolute_links:
+                url = urlparse(l)
+                qs = parse_qs(url.query)
+                if 'url' not in qs:
+                    continue
+                new_link = qs['url'][0]
+                # if self.verify_url(new_link):
+                #     full_linkset.append(new_link)
```
Comment (TPGamesNL) [inline engines/google.py:147]: Reminder

File: engines/google.py [hunk 3]
```diff
@@ -152,71 +161,58 @@ def post_html(self, url=None, region=None):
 
         if not (region is None):
             if type(region) is numpy.ndarray:
-                multipart = {'encoded_image': ('temp.png', cv2.imencode('.png', region)[1])}
+                png_img = cv2.imencode('.png', region)[1]
+                multipart = {'encoded_image': ('temp.png', png_img)}
+                with open(f'files/temp-{random.randint(0, 999999)}.png', 'wb') as f:
+                    f.write(numpy.array(png_img).tobytes())
```
Comment (TPGamesNL) [inline engines/google.py:167]: Reminder

**Comment by TPGamesNL** [comment]
last TODOs:
- re-add clearbit API
- more future-proof: improve error detection (and fixing by automatically saving HTML files) of stuff like different page format, or cookie consent popups

**Review by ronalddenouden — APPROVED** [review]
Well done on cleaning up and creating a clear superclass for reverseimagesearch. The implementation of Google looks good!

**Pull request merged**